### DISCUSSION TITLE
Prompting the user for program and arguments at once

### DISCUSSION THREAD

**Discussion by LordMZTE** [mfussenegger/nvim-dap#758] [Q&A] [answered]
I want to debug my C++ project, and I currently have this set up to have a function for `program` and for 'args` to prompt the user, but this means the user will be prompted twice for arguments and the binary, which I don't really like. Has anyone figured out how to ask for both at once (split with a space)?

**Comment by mfussenegger** [comment, marked as answer]
It currently requires a bit of a hack, see: https://github.com/mfussenegger/nvim-dap/discussions/597#discussioncomment-3098086

**Reply by LordMZTE** [reply]
Ah, thank you!

**Reply by mfussenegger** [reply]
I also added support for metatables with `__call` on configurations, see https://github.com/mfussenegger/nvim-dap/pull/768/files
That should make it less of a hack as it avoids forward references or mutating properties
